### ISSUE TITLE
Problem with client autoconfiguration

### ISSUE THREAD

**Issue by ramses-gomez** [yidongnan/grpc-spring-boot-starter#194]
Not sure if I'm missing something bout for some reason the grpc tracing interceptor is not being created and I can't see why:

GrpcClientAutoConfiguration.TraceClientAutoConfiguration:
      Did not match:
         - @ConditionalOnBean (types: brave.grpc.GrpcTracing; SearchStrategy: all) did not find any beans of type brave.grpc.GrpcTracing (OnBeanCondition)
      Matched:
         - @ConditionalOnProperty (spring.sleuth.scheduled.enabled) matched (OnPropertyCondition)

However, I can see the bean created:

GrpcCommonTraceAutoConfiguration matched:
      - @ConditionalOnClass found required classes 'brave.Tracing', 'brave.grpc.GrpcTracing' (OnClassCondition)
      - @ConditionalOnProperty (spring.sleuth.scheduled.enabled) matched (OnPropertyCondition)

The only thing I can think of is that the bean is created after the TraceClientAutoConfiguration is being evaluated but I don't see how is this possible.

Am I missing something here?

The Server side is being created successfully.

Any help is appreciated,

Thanks

**Comment by ramses-gomez** [comment]
I'm using version 2.2.1.Release and spring boot version 2.1.2.Release

**Comment by ST-DDT** [comment]
I will look into it.

**Comment by ST-DDT** [comment]
I could reproduce this issue locally and created a PR to fix it. The fix will be included in the next version 2.3.0.

Until then you can bypass the issue by manually creating the skipped bean:

````java
@Bean
public GlobalClientInterceptorConfigurer globalTraceClientInterceptorConfigurerAdapter(GrpcTracing grpcTracing) {
    return registry -> registry.addClientInterceptors(grpcTracing.newClientInterceptor());
}
````

Maybe you have to wrap it in its own auto configuration class as shown in the linked PR.
